src/oslogin_utils.cc

```diff
@@ -554,31 +569,35 @@ bool ParseJsonToUsers(const string& json, std::vector<string>* result) {
 }
 
 bool ParseJsonToGroups(const string& json, std::vector<Group>* result) {
-  json_object* root = NULL;
-  root = json_tokener_parse(json.c_str());
-  if (root == NULL) {
-    return false;
-  }
-
   bool ret = false;
 
-  json_object* groups = NULL;
+  json_object* root = ParseJsonRoot(json);
+  if (root == NULL) {
+    return ret;
+  }
+  json_object* groups;
+  json_type groupType;
   if (!json_object_object_get_ex(root, "posixGroups", &groups)) {
+    SysLogErr("failed to parse POSIX groups from \"%s\"", json);
     goto cleanup;
   }
-  if (json_object_get_type(groups) != json_type_array) {
+  groupType = json_object_get_type(groups);
+  if (groupType != json_type_array) {
+    SysLogErr("parsed unexpected type for field \"posixGroups\"; "
+              "want a list, got %s", groupType);
     goto cleanup;
   }
   for (int idx = 0; idx < (int)json_object_array_length(groups); idx++) {
     json_object* group = json_object_array_get_idx(groups, idx);
 
     json_object* gid;
+    json_object* name;
     if (!json_object_object_get_ex(group, "gid", &gid)) {
+      SysLogErr("failed to parse gid from group %s", json_object_get_string(group));
       goto cleanup;
     }
-
-    json_object* name;
```

Comment:
huuum, I see you are intentionally grouping these variable's declaration, why?

Reply:
Old C89 habits die hard. ;)

Really, it has to do with the line spacing: see how lines 583 through 602 serve to declare and then define the `gid` and `name` variables. This "stanza," surrounded by empty lines, is like a little self-contained pseudo-function. So the stanza reads to me like "here are two variables I'm going to set. And now here is how I set them."

I'm happy to move the declaration down a bit, though; I also usually prefer to declare things right where they're used, too. I've changed all instances of this pattern to be declared later, as you suggest.